lib/topological_inventory/openshift/operations/core/service_catalog_client.rb

```diff
@@ -1,64 +1,93 @@
           end
 
           def order_service_plan(plan_name, service_offering_name, additional_parameters)
-            payload = ServicePlanClient.new.build_payload(plan_name, service_offering_name, additional_parameters)
-            response = external_request(:post, order_service_plan_url, payload)
-            JSON.parse(response.body)
+            connection = connection_manager.connect(
+              "servicecatalog", host: default_endpoint.host, token: authentication.password, verify_ssl: verify_ssl_mode)
+
+            payload = build_payload(plan_name, service_offering_name, additional_parameters)
+
+            service_instance = connection.create_service_instance(payload)
+
+            logger.info("Waiting for #{service_instance.metadata.name} to provision...")
+            service_instance = wait_for_provision_complete(connection, service_instance)
+            logger.info("Waiting for #{service_instance.metadata.name} to provision...Complete")
+
+            service_instance
           end
 
           private
 
-          def order_service_plan_url
-            base_url_path = URI::Generic.build(
-              :scheme => @default_endpoint.scheme,
-              :host   => @default_endpoint.host,
-              :port   => @default_endpoint.port,
-              :path   => @default_endpoint.path
-            ).to_s
-            URI.join(base_url_path, "apis/servicecatalog.k8s.io/v1beta1/namespaces/default/serviceinstances").to_s
-          end
+          def build_payload(service_plan_name, service_offering_name, order_parameters)
+            # We need to not send empty strings in case the parameter is generated
+            # More details are explained in the comment in the OpenShift web catalog
+            # https://github.com/openshift/origin-web-catalog/blob/4c5cb3ee1ae0061ed28fc6190a0f8fff71771122/src/components/order-service/order-service.controller.ts#L442
+            safe_params = order_parameters["service_parameters"].delete_blanks
 
-          def external_request(method, url, payload, headers = generic_headers)
-            request_options = {
-              :method     => method,
-              :url        => url,
-              :headers    => headers,
-              :verify_ssl => verify_ssl_mode,
-              :payload    => payload
+            {
+              :metadata   => {
+                :name      => "#{service_offering_name}-#{SecureRandom.uuid}",
+                :namespace => order_parameters["provider_control_parameters"]["namespace"]
+              },
+              :spec       => {
+                :clusterServiceClassExternalName => service_offering_name,
+                :clusterServicePlanExternalName  => service_plan_name,
+                :parameters                      => safe_params
+              }
             }
+          end
+
+          def wait_for_provision_complete(connection, service_instance)
+            loop do
```

Comment:
So we'll need to do sync waiting, until we have some tool for async workers, right?